utilities/transactions/pessimistic_transaction.cc

```diff
@@ -515,7 +515,9 @@ Status PessimisticTransaction::LockBatch(WriteBatch* batch,
 // the snapshot time.
 Status PessimisticTransaction::TryLock(ColumnFamilyHandle* column_family,
                                        const Slice& key, bool read_only,
-                                       bool exclusive, bool skip_validate) {
+                                       bool exclusive, const bool do_validate,
+                                       const bool assume_exclusive_tracked) {
+  assert(!assume_exclusive_tracked || (!do_validate && exclusive));
```

Comment:
I think Manuel mentioned acquiring read locks should be able to set do_validate = false if we want to retrieve the value with snapshot set to nullptr.

Reply:
Correct. The latest PR is updated with manuel's comment. Let me know if the latest is correct.